capacity-mode-evaluator/README.md

```diff
@@ -0,0 +1,136 @@
+    ```
+
+    Add the `--debug` flag to save metrics and estimates as CSV files in the `output` folder.
+
+3. The output files will be saved in the `output` folder.
+
+## Output
+
+Check the generated files in the `output` folder. [Here](static/analysis_summary.csv) is an example output
+
+**Note:** The information generated is applicable only to the period specified by `DAYS_LOOK_BACK` and is provided on separate lines for the table and relevant index for both `ReadCapacityUnits` and `WriteCapacityUnits`.
+
+The **`analysis_summary.csv`** file contains the following columns,
+
+- `index_name`: The name of the index associated table in the analysis.
+- `base_table_name`: The name of the table associated with the analysis.
+- `class`: Storage class of the table associated with the analysis.
+- `metric_name`: The name of the metric associated with the analysis.
+- `est_provisioned_cost`: The estimated cost of the Provisioned mode based on the table's simulated usage.
+- `current_provisioned_cost`: If table capacity mode is provisioned, the table's current Provisioned cost.
+- `ondemand_cost`: The cost of using on-demand  mode for the table.
+- `recommended_mode`: The recommended capacity mode for the table.
+- `current_mode`: The table's current capacity mode.
+- `status`: The status of the table or index if `Optimized` or `Not Optimized` based on the analysis.
+- `savings_pct`: The estimated percentage of cost savings considering the recommended capacity mode.
+- `number_of_days`: The number of days in the lookback period for the analysis.
+- `current_min_capacity`: If table capacity mode is provisioned and autoscaling is enabled, the table's current minimum capacity units.
+- `simulated_min_capacity`: The minimum capacity units based on what the provisioned capacity mode is simulated.
+- `current_max_capacity`: If table capacity mode is provisioned and autoscaling is enabled, the table's current maximum capacity units.
+- `simulated_max_capacity`: The maximum capacity units based on what the provisioned capacity mode is simulated.
+- `current_target_utilization`: If table capacity mode is provisioned and autoscaling is enabled, the table's current target utilization.
+- `simulated_target_utilization`: The target utilization based on what the provisioned capacity mode is simulated.
+- `current_cost`: The table's current cost for period analyzed.
+- `recommended_cost`: The table's estimated cost for period analyzed considering the recommended capacity mode.
+- `autoscaling_enabled`: The Table's current Autoscaling Status.
+- `Note`: When changing the capacity mode from On-Demand to Provisioned or vice versa, consider the tools limitations such as lack of consideration for scaling delays and 14 days lookback period.
```

Comment:
Suggestion:

**_Note:_**

The analysis provided in this script compares your table consumption and simulates cost using different parameters. This tool does not have access to your contextual information, business requirements or organization best practices. When changing your capacity mode from on-demand to provisioned based on the results, remember there were some assumptions made: 
 - The analysis window is 14 days.
 - Auto-scaling responds instantaneously. (In reality, Auto scaling service might take 4 mins to provision new table capacity depending on your increase conditions).